src/index.js

```diff
@@ -230,11 +230,12 @@ class QRCodeCanvas extends React.PureComponent<QRProps, {imgLoaded: boolean}> {
     this.update();
   }
 
-  componentWillReceiveProps(nextProps) {
+  static getDerivedStateFromProps(nextProps, prevState) {
     const currentSrc = this.props.imageSettings?.src;
     const nextSrc = nextProps.imageSettings?.src;
+
     if (currentSrc !== nextSrc) {
-      this.setState({imgLoaded: false});
+      this.setState({imgLoaded: false})
```

Comment:
I think it should be renamed to `UNSAFE_componentWillReceiveProps` as its recommended in the [official update post](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html). Also, please don't remove the semicolon after the `setState()`

Reply:
I've updated it to use componentDidUpdate instead. Do take a look

Reply:
@zpao Surely you know better than me that the suggested change to `getDerivedStateFromProps ` is not doable as written.
But we can store the `prevProps` in state and derive the state accordingly when new props arrive:

```js
   static getDerivedStateFromProps(props, state) {
    const currentSrc = state.imageSrc;
    const nextSrc = props.imageSettings?.src;

    if (currentSrc !== nextSrc) {
      return {imgSrc: nextSrc, imgLoaded: false};
    }
  }
```

Could you pls publish some fix for this issue? using the mentioned above approach or just renaming the method `UNSAFE_componentWillReceiveProps` at least for the >= v16.3.0.